src/main.rs

```diff
@@ -97,19 +96,112 @@ enum ClientEvent {
         avatar: String,
         code: String,
     },
+}
+
+#[derive(Deserialize)]
+#[serde(tag = "tag")]
+enum ClientEventGame {
     /// When the admin of the room starts the game
-    StartRound { code: String },
+    StartRound,
     /// When Client answers to a question
-    Answer { code: String, vote: Vote },
+    Answer { vote: Vote },
 }
 
 lazy_static::lazy_static! {
-    static ref ROOMS: Mutex<HashMap<String,Room>> = Mutex::new(HashMap::new());
+    static ref ROOMS: Mutex<HashMap<String, Room<GameRoom>>> = Mutex::new(HashMap::new());
+    static ref QUESTIONS: Vec<Prompt> = {
+        // TODO: c pa opti
+        let content = std::fs::read_to_string("questions.ron").expect("Error while reading question.ron file");
+        let questions: Vec<Prompt> = ron::from_str(&content).expect("Error while parsing questions");
+
+        println!("Read {} questions ({} tags)", questions.len(), questions.iter().filter(|x| match x {
+            Prompt::Tag(_, _) => true,
+            _ => false,
+        }).count());
+
+        questions
+    };
 }
 
-fn send_msg(out: &ws::Sender, msg: &ServerEvent) -> ws::Result<()> {
-    let msg = serde_json::to_string(&msg).unwrap();
-    out.send(ws::Message::Text(msg))
+impl From<&ServerEvent<'_>> for Message {
+    fn from(event: &ServerEvent<'_>) -> Self {
+        Message::Text(serde_json::to_string(event).unwrap())
+    }
+}
+
+impl RoomHandler for GameRoom {
+    type Guest = String;
+
+    fn on_join(&mut self, cx: Context<Self::Guest>) -> ResultRelocation {
+        // Send an update to each other player
+        cx.broadcast(&ServerEvent::RoomUpdate {
+            players: self.players.clone(),
+        })?;
+
+        cx.send(&ServerEvent::OnRoomJoin {
+            players: self.players.clone(),
+        })?;
+
+        Ok(None)
+    }
+
+    fn on_message(&mut self, cx: Context<Self::Guest>, msg: Message) -> ResultRelocation {
+        use ServerEvent::*;
+
+        let msg = match msg {
+            Message::Text(s) => s,
+            _ => unreachable!(),
+        };
+
+        let evt = match serde_json::from_str(&msg) {
+            Ok(msg) => msg,
+            _ => return Ok(None),
+        };
+
+        match evt {
+            ClientEventGame::StartRound => {
+                self.votes.clear();
+                if self.questions_count > 9 {
+                    cx.broadcast(&GameOver)?;
```

Comment:
Does `broadcast` sends the message to the client that sent the "StartRound" event too? Or just to other players?

Reply:
Everyone.

I know that it didn't do that before but there's no easy way to send a message to "everyone except ...". And it seems to work fine like that. If that's a problem for a specific edge case or some future plan, I can always implement the feature in ws-hotel and change this line.

Reply:
Wait no, I don't remember actually, I was thinking about main.rs:132 in e5bd9c7429d55b8e8106e3d8a1c46935826156e0, where I'm sure the broadcast was not meant for the person initiating it.